src/test/java/hudson/scm/IntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+                    while (true) {
+                        final Socket s;
+                        try {
+                            s = sock.accept();
+                        } catch (SocketException x) {
+                            // Socket closed?
+                            break;
+                        }
+                        LOGGER.info("accepted client connection");
+                        ProcessBuilder pb = new ProcessBuilder("cvs", "-f", "--allow-root=" + repo, "pserver");
+                        final Process server = pb.start();
+                        new Thread("printing errors") {
+                            @Override public void run() {
+                                try {
+                                    copy(server.getErrorStream(), System.err);
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("sending output") {
+                            @Override public void run() {
+                                try {
+                                    copy(server.getInputStream(), /*new TeeOutputStream(*/s.getOutputStream()/*, System.err)*/);
+                                } catch (SocketException x) {
+                                    // Broken pipe? Ignore.
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("accepting input") {
+                            @Override public void run() {
+                                try {
+                                    InputStream is = s.getInputStream();
+                                    OutputStream os = server.getOutputStream();
+                                    copy(is, /*new TeeOutputStream(*/os/*, System.err)*/);
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("wait for exit") {
+                            @Override public void run() {
+                                try {
+                                    int r = server.waitFor();
+                                    if (r != 0) {
+                                        LOGGER.log(Level.INFO, "server exited with status {0}", r);
+                                    }
+                                } catch (Exception x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        };
+                    }
+                } catch (IOException x) {
+                    x.printStackTrace();
```

Comment:
Although I agree with you that tests should be treated as production code, I usually do some exceptions in logger facilities. In the past I have always used loggers in test (basically JUL), but we end up by having a lot of problems with users who use those classes, because maybe don't have the logger correctly configured, there are some loggers in classpath that makes everything not log what you really want to see, ..., or maybe you have on classpath different kind of loggers which at end makes everything a headache. To avoid this we started to use on tests System.out and e.printstacktrace to receive quick feedback of what is failing in test.

As I said I like to treat tests as first-class citizen but at the same time I want to make life easier for everyone, and throwing some printStackTraces it is not bad at all IMO.